core/kover/learning/common/models.py

```diff
@@ -21,11 +22,80 @@
 
 conjunction = "conjunction"
 disjunction = "disjunction"
+scm = "scm"
+cart = "cart"
 
 class BaseModel(object):
     def __init__(self):
-        self.rules = []
+        self.rules = None
         super(BaseModel, self).__init__()
+        
+    def predict(self, X):
+        raise NotImplementedError()
+        
+    @property
+    def learner(self):
+        raise NotImplementedError()
+        
+    def __str__(self):
+        return self._to_string()
+        
+class CART_Model(BaseModel):
+    def __init__(self, class_tags=None):
+        super(CART_Model, self).__init__()
+        self.decision_tree = None
+        self.class_tags = class_tags
+        
+    def add_decision_tree(self, tree):
```

Comment:
You don't need a function for this. You won't be adding many decision trees. You could do the same by doing model.decision_tree = tree.